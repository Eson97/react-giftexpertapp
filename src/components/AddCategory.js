import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = (({ setCategories }) => {

    const [inputValue, setInputValue] = useState('Nueva categorÃ­a');

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length > 0) {
            setCategories(cat => [inputValue, ...cat]);
            setInputValue('');
        }
    }

    const handleFocus = () => {
        setInputValue('');
    }

    const handleBlur = () =>{
        setInputValue('Nueva categoria');
    }

    return (
        <form onSubmit={handleSubmit} onFocus={handleFocus} onBlur={handleBlur}>
            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )

})

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
