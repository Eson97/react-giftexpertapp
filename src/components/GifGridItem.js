import React from 'react'



export const GifGridItem = ({title,url}) => {
    
    return (
        <div className="card animate__animated animate__fadeInLeft">
            <img src={url} alt={title}/>
            <p>{title}</p>
            <iframe title="share-button" src={`https://www.facebook.com/plugins/share_button.php?href=${url}&layout=button&size=large&width=103&height=28&appId`} width="103" height="28" style={{border:"none",overflow:"hidden",margin:"10px"}} scrolling="no" frameborder="0" allowfullscreen="true" allow="autoplay; clipboard-write; encrypted-media; picture-in-picture; web-share"></iframe>
        </div>
    )
}
