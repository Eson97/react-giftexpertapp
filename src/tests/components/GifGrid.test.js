import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs'); //simula la llamada al archivo y se manipula la informacion que este contiene

describe('test on <GifGrid />', () => {

    const category = 'cat';

    test('should show correctly', () => {

        useFetchGifs.mockReturnValue({ //simulamos el hook en su estado base
            data: [],
            loading: true
        })

        const wrapper = shallow(<GifGrid category={category} />);

        expect(wrapper).toMatchSnapshot();

    })

    test('should show items when imgs are loaded from useFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cat.jpg',
            title: 'Cat'
        }]

        useFetchGifs.mockReturnValue({ //simulamos el hook con informacion
            data: gifs,
            loading: false
        })

        const wrapper = shallow(<GifGrid category={category} />);

        //expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

    })


})
