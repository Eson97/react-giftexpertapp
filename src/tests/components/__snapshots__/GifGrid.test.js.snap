// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on <GifGrid /> should show correctly 1`] = `
<Fragment>
  <h3>
     
    cat
     
  </h3>
  <p>
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
